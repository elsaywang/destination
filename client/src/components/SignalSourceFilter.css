:local(.heading) {
    display: block;
    padding-left: 29px;
    font-size: 1.2em;
    font-weight: lighter;
}

:local(.signalSource) {
    padding: 0 20px;
}
