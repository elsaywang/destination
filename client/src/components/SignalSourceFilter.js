import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from '@react/react-spectrum/Select';
import styles from './SignalSourceFilter.css';
import {
    getSignalSourceFilterPlaceholderText,
    getSignalSourcesOptions,
} from '../utils/signalSourceOptions';

const SignalSourceFilter = ({
    disabled,
    sourceType,
    onSignalSourceSelect,
    signalSources,
    selectedSignalSource,
}) => {
    const filterPlaceholder = `Filter by ${getSignalSourceFilterPlaceholderText(sourceType)}`;
    const options = getSignalSourcesOptions(signalSources, sourceType);

    return (
        <Select
            data-test={`${sourceType.toLowerCase()}-signal-source-filter`}
            className={styles.signalSourcesFilter}
            placeholder={filterPlaceholder}
            onChange={onSignalSourceSelect}
            options={options}
            className={styles.signalSourceSelect}
            flexible
            disabled={disabled}
            value={selectedSignalSource}
        />
    );
};

SignalSourceFilter.propTypes = {
    disabled: PropTypes.bool.isRequired,
    sourceType: PropTypes.string.isRequired,
    onSignalSourceSelect: PropTypes.func.isRequired,
    signalSources: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string,
            dataSourceId: PropTypes.number,
            suite: PropTypes.string,
        }),
    ).isRequired,
    selectedSignalSource: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default SignalSourceFilter;
