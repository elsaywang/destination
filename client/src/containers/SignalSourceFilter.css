:local(.heading) {
    display: block;
    padding: 0 20px;
    font-size: 1.2em;
    font-weight: lighter;
}

:local(.tabList) {
    padding: 20px 40px 20px 20px;
}

:local(.tabList) .spectrum-TabList-item {
    margin: 0;
    padding: 0;
    border: none;
}

:local(.tabList) .is-selected span {
    font-weight: bold;
}
