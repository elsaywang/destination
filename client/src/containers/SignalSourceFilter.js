import React, { Fragment } from 'react';
import { Tab, TabList } from '@react/react-spectrum/TabList';
import Heading from '@react/react-spectrum/Heading';
import styles from './SignalSourceFilter.css';

function SignalSourceFilter({ counts, handleSignalSourceChange }) {
    return (
        <Fragment>
            <Heading size={3} className={styles.heading}>
                Filter By Signal Source:
            </Heading>
            <TabList
                className={styles.tabList}
                orientation="vertical"
                onChange={handleSignalSourceChange}>
                <Tab>All ({counts.all})</Tab>
                <Tab>Adobe Analytics ({counts.adobeAnalytics})</Tab>
                <Tab>Actionable Log Files ({counts.actionableLogFiles})</Tab>
                <Tab>General Online Data ({counts.generalOnlineData})</Tab>
                <Tab>Onboarded Records ({counts.onboardedRecords})</Tab>
            </TabList>
        </Fragment>
    );
}

export default SignalSourceFilter;
