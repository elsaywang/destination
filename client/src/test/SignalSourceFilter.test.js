import React from 'react';
import { shallow } from 'enzyme';
import SignalSourceFilter from '../components/SignalSourceFilter';
import Heading from '@react/react-spectrum/Heading';
import { Tab, TabList } from '@react/react-spectrum/TabList';
import getSignalSourceOptions from '../constants/getSignalSourceOptions';

describe('<SignalSourceFilter /> component', () => {
    const mockFn = jest.fn();
    const initialState = {
        counts: {
            all: 72093,
            adobeAnalytics: 34300,
            actionableLogFiles: 359,
            generalOnlineData: 27,
            onboardedRecords: 37407,
        },
        filter: 'all',
    };
    let wrapper = shallow(
        <SignalSourceFilter
            handleSignalSourceChange={mockFn}
            counts={initialState.counts}
            filter={initialState.filter}
        />,
    );
    const signalSourceOptions = getSignalSourceOptions(initialState.counts);

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Heading /> component', () => {
            expect(wrapper.find(Heading).exists()).toBe(true);
        });

        it('renders <TabList /> component with correct number of <Tab />s', () => {
            expect(wrapper.find(TabList).exists()).toBe(true);
            expect(wrapper.find(Tab).length).toBe(signalSourceOptions.length);
        });

        it('renders correctly selected <Tab /> component when props.filter change', () => {
            const filter = 'adobeAnalytics';
            const optionMatchingFilter = signalSourceOptions.find(option => option.value === filter)
                .label;

            wrapper.setProps({ filter });

            expect(
                wrapper
                    .find(Tab)
                    .filterWhere(tab => tab.props().selected)
                    .html(),
            ).toContain(optionMatchingFilter);
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Tab /> component with correct label when props.count change', () => {
            wrapper
                .find(Tab)
                .map((tab, index) =>
                    expect(tab.html()).toContain(signalSourceOptions[index].label),
                );

            const newState = {
                counts: {
                    all: 1100,
                    adobeAnalytics: 1000,
                    actionableLogFiles: 100,
                    generalOnlineData: 0,
                    onboardedRecords: 0,
                },
            };
            const newSignalSourceOptions = getSignalSourceOptions(newState.counts);

            wrapper.setProps({
                ...newState,
            });

            wrapper
                .find(Tab)
                .map((tab, index) =>
                    expect(tab.html()).toContain(newSignalSourceOptions[index].label),
                );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
