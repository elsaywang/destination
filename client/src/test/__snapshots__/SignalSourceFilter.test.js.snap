// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignalSourceFilter /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalSourceFilter
    counts={
      Object {
        "actionableLogFiles": 359,
        "adobeAnalytics": 34300,
        "all": 72093,
        "generalOnlineData": 27,
        "onboardedRecords": 37407,
      }
    }
    filter="all"
    handleSignalSourceChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Heading
          className={undefined}
          size={3}
        >
          Filter By Signal Source:
        </Heading>,
        <TabList
          className={undefined}
          onChange={[MockFunction]}
          orientation="vertical"
          value="all"
        >
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={true}
          >
            All (72093)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Adobe Analytics (34300)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Actionable Log Files (359)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            General Online Data (27)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Onboarded Records (37407)
          </Tab>
        </TabList>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Filter By Signal Source:",
          "className": undefined,
          "size": 3,
        },
        "ref": null,
        "rendered": "Filter By Signal Source:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              All (72093)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Adobe Analytics (34300)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>,
          ],
          "className": undefined,
          "onChange": [MockFunction],
          "orientation": "vertical",
          "value": "all",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "all",
            "nodeType": "class",
            "props": Object {
              "children": "All (72093)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": true,
            },
            "ref": null,
            "rendered": "All (72093)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "adobeAnalytics",
            "nodeType": "class",
            "props": Object {
              "children": "Adobe Analytics (34300)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Adobe Analytics (34300)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "actionableLogFiles",
            "nodeType": "class",
            "props": Object {
              "children": "Actionable Log Files (359)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Actionable Log Files (359)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "generalOnlineData",
            "nodeType": "class",
            "props": Object {
              "children": "General Online Data (27)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "General Online Data (27)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "onboardedRecords",
            "nodeType": "class",
            "props": Object {
              "children": "Onboarded Records (37407)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Onboarded Records (37407)",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Heading
            className={undefined}
            size={3}
          >
            Filter By Signal Source:
          </Heading>,
          <TabList
            className={undefined}
            onChange={[MockFunction]}
            orientation="vertical"
            value="all"
          >
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              All (72093)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Adobe Analytics (34300)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>
          </TabList>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Filter By Signal Source:",
            "className": undefined,
            "size": 3,
          },
          "ref": null,
          "rendered": "Filter By Signal Source:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={true}
              >
                All (72093)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Adobe Analytics (34300)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Actionable Log Files (359)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                General Online Data (27)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Onboarded Records (37407)
              </Tab>,
            ],
            "className": undefined,
            "onChange": [MockFunction],
            "orientation": "vertical",
            "value": "all",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "all",
              "nodeType": "class",
              "props": Object {
                "children": "All (72093)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": true,
              },
              "ref": null,
              "rendered": "All (72093)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "adobeAnalytics",
              "nodeType": "class",
              "props": Object {
                "children": "Adobe Analytics (34300)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Adobe Analytics (34300)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "actionableLogFiles",
              "nodeType": "class",
              "props": Object {
                "children": "Actionable Log Files (359)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Actionable Log Files (359)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "generalOnlineData",
              "nodeType": "class",
              "props": Object {
                "children": "General Online Data (27)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "General Online Data (27)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "onboardedRecords",
              "nodeType": "class",
              "props": Object {
                "children": "Onboarded Records (37407)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Onboarded Records (37407)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SignalSourceFilter /> component rendering renders <Tab /> component with correct label when props.count change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalSourceFilter
    counts={
      Object {
        "actionableLogFiles": 100,
        "adobeAnalytics": 1000,
        "all": 1100,
        "generalOnlineData": 0,
        "onboardedRecords": 0,
      }
    }
    filter="adobeAnalytics"
    handleSignalSourceChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Heading
          className={undefined}
          size={3}
        >
          Filter By Signal Source:
        </Heading>,
        <TabList
          className={undefined}
          onChange={[MockFunction]}
          orientation="vertical"
          value="adobeAnalytics"
        >
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            All (1100)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={true}
          >
            Adobe Analytics (1000)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Actionable Log Files (100)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            General Online Data (0)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Onboarded Records (0)
          </Tab>
        </TabList>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Filter By Signal Source:",
          "className": undefined,
          "size": 3,
        },
        "ref": null,
        "rendered": "Filter By Signal Source:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (1100)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (1000)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (100)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (0)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (0)
            </Tab>,
          ],
          "className": undefined,
          "onChange": [MockFunction],
          "orientation": "vertical",
          "value": "adobeAnalytics",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "all",
            "nodeType": "class",
            "props": Object {
              "children": "All (1100)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "All (1100)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "adobeAnalytics",
            "nodeType": "class",
            "props": Object {
              "children": "Adobe Analytics (1000)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": true,
            },
            "ref": null,
            "rendered": "Adobe Analytics (1000)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "actionableLogFiles",
            "nodeType": "class",
            "props": Object {
              "children": "Actionable Log Files (100)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Actionable Log Files (100)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "generalOnlineData",
            "nodeType": "class",
            "props": Object {
              "children": "General Online Data (0)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "General Online Data (0)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "onboardedRecords",
            "nodeType": "class",
            "props": Object {
              "children": "Onboarded Records (0)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Onboarded Records (0)",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Heading
            className={undefined}
            size={3}
          >
            Filter By Signal Source:
          </Heading>,
          <TabList
            className={undefined}
            onChange={[MockFunction]}
            orientation="vertical"
            value="adobeAnalytics"
          >
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (1100)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (1000)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (100)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (0)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (0)
            </Tab>
          </TabList>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Filter By Signal Source:",
            "className": undefined,
            "size": 3,
          },
          "ref": null,
          "rendered": "Filter By Signal Source:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                All (1100)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={true}
              >
                Adobe Analytics (1000)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Actionable Log Files (100)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                General Online Data (0)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Onboarded Records (0)
              </Tab>,
            ],
            "className": undefined,
            "onChange": [MockFunction],
            "orientation": "vertical",
            "value": "adobeAnalytics",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "all",
              "nodeType": "class",
              "props": Object {
                "children": "All (1100)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "All (1100)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "adobeAnalytics",
              "nodeType": "class",
              "props": Object {
                "children": "Adobe Analytics (1000)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": true,
              },
              "ref": null,
              "rendered": "Adobe Analytics (1000)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "actionableLogFiles",
              "nodeType": "class",
              "props": Object {
                "children": "Actionable Log Files (100)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Actionable Log Files (100)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "generalOnlineData",
              "nodeType": "class",
              "props": Object {
                "children": "General Online Data (0)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "General Online Data (0)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "onboardedRecords",
              "nodeType": "class",
              "props": Object {
                "children": "Onboarded Records (0)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Onboarded Records (0)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SignalSourceFilter /> component rendering renders correctly selected <Tab /> component when props.filter change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalSourceFilter
    counts={
      Object {
        "actionableLogFiles": 359,
        "adobeAnalytics": 34300,
        "all": 72093,
        "generalOnlineData": 27,
        "onboardedRecords": 37407,
      }
    }
    filter="adobeAnalytics"
    handleSignalSourceChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Heading
          className={undefined}
          size={3}
        >
          Filter By Signal Source:
        </Heading>,
        <TabList
          className={undefined}
          onChange={[MockFunction]}
          orientation="vertical"
          value="adobeAnalytics"
        >
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            All (72093)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={true}
          >
            Adobe Analytics (34300)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Actionable Log Files (359)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            General Online Data (27)
          </Tab>
          <Tab
            disabled={false}
            invalid={false}
            renderChildren={true}
            selected={false}
          >
            Onboarded Records (37407)
          </Tab>
        </TabList>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Filter By Signal Source:",
          "className": undefined,
          "size": 3,
        },
        "ref": null,
        "rendered": "Filter By Signal Source:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (72093)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (34300)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>,
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>,
          ],
          "className": undefined,
          "onChange": [MockFunction],
          "orientation": "vertical",
          "value": "adobeAnalytics",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "all",
            "nodeType": "class",
            "props": Object {
              "children": "All (72093)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "All (72093)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "adobeAnalytics",
            "nodeType": "class",
            "props": Object {
              "children": "Adobe Analytics (34300)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": true,
            },
            "ref": null,
            "rendered": "Adobe Analytics (34300)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "actionableLogFiles",
            "nodeType": "class",
            "props": Object {
              "children": "Actionable Log Files (359)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Actionable Log Files (359)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "generalOnlineData",
            "nodeType": "class",
            "props": Object {
              "children": "General Online Data (27)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "General Online Data (27)",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "onboardedRecords",
            "nodeType": "class",
            "props": Object {
              "children": "Onboarded Records (37407)",
              "disabled": false,
              "invalid": false,
              "renderChildren": true,
              "selected": false,
            },
            "ref": null,
            "rendered": "Onboarded Records (37407)",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Heading
            className={undefined}
            size={3}
          >
            Filter By Signal Source:
          </Heading>,
          <TabList
            className={undefined}
            onChange={[MockFunction]}
            orientation="vertical"
            value="adobeAnalytics"
          >
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              All (72093)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={true}
            >
              Adobe Analytics (34300)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Actionable Log Files (359)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              General Online Data (27)
            </Tab>
            <Tab
              disabled={false}
              invalid={false}
              renderChildren={true}
              selected={false}
            >
              Onboarded Records (37407)
            </Tab>
          </TabList>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Filter By Signal Source:",
            "className": undefined,
            "size": 3,
          },
          "ref": null,
          "rendered": "Filter By Signal Source:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                All (72093)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={true}
              >
                Adobe Analytics (34300)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Actionable Log Files (359)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                General Online Data (27)
              </Tab>,
              <Tab
                disabled={false}
                invalid={false}
                renderChildren={true}
                selected={false}
              >
                Onboarded Records (37407)
              </Tab>,
            ],
            "className": undefined,
            "onChange": [MockFunction],
            "orientation": "vertical",
            "value": "adobeAnalytics",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "all",
              "nodeType": "class",
              "props": Object {
                "children": "All (72093)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "All (72093)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "adobeAnalytics",
              "nodeType": "class",
              "props": Object {
                "children": "Adobe Analytics (34300)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": true,
              },
              "ref": null,
              "rendered": "Adobe Analytics (34300)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "actionableLogFiles",
              "nodeType": "class",
              "props": Object {
                "children": "Actionable Log Files (359)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Actionable Log Files (359)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "generalOnlineData",
              "nodeType": "class",
              "props": Object {
                "children": "General Online Data (27)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "General Online Data (27)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "onboardedRecords",
              "nodeType": "class",
              "props": Object {
                "children": "Onboarded Records (37407)",
                "disabled": false,
                "invalid": false,
                "renderChildren": true,
                "selected": false,
              },
              "ref": null,
              "rendered": "Onboarded Records (37407)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
